### PR TITLE
Makefile: alias `build` target as `neo-go`

### LINKED ISSUE #2529: synchronize common make targets across all projects
There should be a list of common make targets that should be the same for all projects.
For example:
- `make bin` makes binaries using host env
- `make docker/bin` -- makes binaries using clean docker env.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Follow the https://github.com/nspcc-dev/neo-go/pull/2537#issuecomment-1147468063.

Close #2529.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
> We better define some `$BIN` (or `$BINARY` and make current one `BINARY_PATH`) and reuse it everywhere.

Coud you see the updated version? I'm not sure, is it what we want.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,8 @@
 BRANCH = "master"
 REPONAME = "neo-go"
 NETMODE ?= "privnet"
-BINARY=./bin/neo-go$(shell go env GOEXE)
+BINARY ?= build
```
Comment (roman-khimov) [inline Makefile:4]: BINARY=neo-go

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
+BINARY_PATH=./bin/neo-go$(shell go env GOEXE)
```
Comment (roman-khimov) [inline Makefile:5]: BINARY_PATH=./bin/$(BINARY)

File: Makefile [hunk 3]
```diff
@@ -22,17 +23,15 @@ IMAGE_REPO=nspccdev/neo-go
 
 # All of the targets are phony here because we don't really use make dependency
 # tracking for files
-.PHONY: build bin deps image image-latest image-push image-push-latest check-version clean-cluster push-tag \
+.PHONY: $(BINARY) deps image image-latest image-push image-push-latest check-version clean-cluster push-tag \
```
Comment (roman-khimov) [inline Makefile:26]: $(BINARY) build

File: Makefile [hunk 4]
```diff
@@ -22,17 +23,15 @@ IMAGE_REPO=nspccdev/neo-go
...
 	test vet lint fmt cover
 
-build: deps
+$(BINARY): deps
```
Comment (roman-khimov) [inline Makefile:29]: build: $(BINARY)

File: Makefile [hunk 5]
```diff
@@ -22,17 +23,15 @@ IMAGE_REPO=nspccdev/neo-go
...
 	@echo "=> Building binary"
 	@set -x \
 		&& export GOGC=off \
 		&& export CGO_ENABLED=0 \
-		&& go build -trimpath -v -ldflags $(BUILD_FLAGS) -o ${BINARY} ./cli/main.go
-
-bin: build
```
Comment (roman-khimov) [inline Makefile:35]: $(BINARY): deps
 ....

**Comment by AnnaShaleva** [comment]
Got it, fixed.

**Comment by roman-khimov** [comment]
Can we also get `docker/$(BINARY)` that will never be used?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**